agent/component/core.go

```diff
@@ -300,31 +268,6 @@ func (c ComponentCore) BulkSubscribe(bCtx *env.BubblyContext) ([]*nats.Subscript
 	return nSubs, nil
 }
 
-// checkConnection checks that the Component has a valid connection to the
-// NATS server prior to a subscription/publication. If not,
-// it attempts to connect to the NATS server.
-func (c ComponentCore) checkConnection(bCtx *env.BubblyContext) error {
```

Comment:
Happy to see this go. However, what I would ask is: does this mean that at the time of request/publish we are sure the `Component` has a connection of status `nats.CONNECTED`? It seems that we establish such at `Component` instantiation, but can an `EncodedConn` terminate after that point without leading to a failure? I do not know the functionality of the `EncodedConn` well enough to be sure. If we aren't checking this prior to publishing/requesting, are we introducing a regression in our ability to handle connection drops etc?